hazelcast/src/main/java/com/hazelcast/partition/impl/InternalPartitionServiceImpl.java

```diff
@@ -1254,8 +1252,8 @@ private boolean hasOnGoingMigrationMaster(Level level) {
 
     @Override
     public boolean hasOnGoingMigrationLocal() {
-        return !activeMigrations.isEmpty() || !migrationQueue.isEmpty()
-                || migrationThread.isMigrating();
+        return !activeMigrations.isEmpty() || migrationQueue.isNonEmpty()
+                || migrationQueue.hasMigrations();
```

Comment:
Only master node knows `migrationQueue`, on other nodes it's always empty. `activeMigrations ` check is required for non-master nodes.